File: abc360b/abc360b.hs
{-# LANGUAGE OverloadedStrings #-}
import Data.List (transpose)
import Data.Bool (bool)

main :: IO ()
main = solve <$> f >>= putStrLn
  where
    f = words <$> getLine

solve :: [String] -> String
solve [s, t] = bool "No" "Yes" $ t `elem` xs
  where
    los = length s
    xs = do
      w <- [1 .. los - 1]
      tp w s

tp :: Int -> String -> [String]
tp w s = transpose $ f s
  where
    f xs =
      let (as, bs) = splitAt w xs in
        if null bs then [as] else as : f bs

File: abc360b/abc360b.ml
open Printf
open Scanf

let solve s t =
  let los = String.length s in
  let rec loop w i =
    if w = los then "No"
    else if i = w then loop (w + 1) 0
    else let l = (los - i + w - 1) / w in
         let ns = String.init l (fun j -> s.[i + j * w]) in
         if ns = t then "Yes"
         else loop w (i + 1) in
  loop 1 0

let () =
  scanf "%s %s " solve |> printf "%s\n"
